packages/preview-middleware/README.md

```diff
@@ -37,7 +37,8 @@ Array of additional application configurations:
 | Option          | Type     | Default Value  | Description                                                                                          |
 | --------------- | -------- | -------------- | ---------------------------------------------------------------------------------------------------- |
 | `target`        | `string` |                | Target path of the additional application                                                            |
-| `local`         | `string` |                | Local path to the folder containing the application                                                  |
+| `local`         | `string` |  `undefined`   | Optional local path to the folder containing the application, if not provided then a `componentId` is required |
+| `componentId`   | `string` |  `undefined`   | Optional `componentId` of the remote app if no local path is provided                                |
```

Comment:
I suppose the goal was to be backwards compatible, but the options and their descriptions are now rather confusing. Using `optional` together with `required` for two options where exactly one should be defined is also misleading and a potential usability issue.

|validity| `local` | `componentId` |  
|---| ------- | ---------------|
|invalid | undefined |  undefined |
|valid | undefined | defined |
|valid | defined |  undefined |
|valid (`componentId` is ignored)| defined | defined |

Reply:
@voicis how about https://github.com/SAP/open-ux-tools/pull/1764/commits/f5a036eea12644314b8f1b0d276ca72f1d89649b? Is that better?

Reply:
Yes, this looks a lot better now. It is clear now from reading the documentation that one of the options must be used and together with the error message pointing out what is wrong, it should be a sufficient solution here.